doc/site/guides/lua-language-server.markdown

```diff
@@ -0,0 +1,244 @@
+---
+layout: default
+title: Lua Language Server
+parent: Guides
+permalink: guides/lua-language-server
+author: rhys-vdw
+---
+
+# Lua Language Server
+
+Lua documentation is generated from `.cpp` files. These are automatically exported to be consumed by games that use Recoil (e.g. BAR). The types are compatible with [Lua Language Server](https://luals.github.io/), providing static analysis and intellisense.
+
+```cpp
```

Comment:
Would be good if the article was sorted into a top half for game-devs, with no mentions of underlying tech (i.e. nothing about cpp), just instructions how to setup the IDE and how to read the Lua side of the docs etc; and a bottom engine-dev section with all the backend info. This is because most people will be interested in one or the other so mixing the two would just confuse.

Reply:
Thanks for the feedback. I'll revise. I'll make two h3s "implications for game dev" and "implications for engine dev"?

Reply:
Actually I'll drop the engine specifics: https://github.com/beyond-all-reason/spring/pull/1775#discussion_r1940287795

Reply:
The running changelog is for gamedevs but anything else is fair game for engine dev internals. If you don't want to rewrite the article then IMO keep as-is, a bit of reordering is not worth losing valuable documentation and can be done by someone else later.

Reply:
Oops thought this was a comment on the changrlog. Yeah I'll split the article.

Reply:
I've split the article: d92689f48e257a23514efeb6248e05b189615714

Engine: https://rhys-vdw.github.io/spring/development/documenting-lua
Game: https://rhys-vdw.github.io/spring/guides/lua-language-server

I think that's a big improvement. Also made some fixes to the content while I was there. I think I'd prefer to merge in this state and iterate on the content now though, if you're comfortable with it as it stands.